### PR TITLE
Re-add batched access checks & replace `j.s.AccessControlException`

### LINKED ISSUE #5923: Replace usages of `java.security.AccessControlException`
`java.security.AccessControlException` is deprecated for removal. We use `ACE` for authz. Can probably easily replace it with our own exception type.

### LINKED ISSUE #5926: Re-add batch authz checks
With #5901 we changed the (internal) interfaces to separate concerns:
* let service-level implementation _push_ entries so that
* resource implementation (think: REST resources) can properly care about page sizes

The change in 5901 was already big-ish and made it necessary to temporarily remove some batch-authz check.

There need to be two follow-ups:
1. unify `*ApiImpl` and `*ApiImplWithAuthorization` classes in `:nessie-services`
2. re-add batch access checks for entries et al
3. address the comments in #5927

### PR DESCRIPTION
**Pull request by snazy**
Fixes #5926
Fixes #5923

### CODE REVIEW

File: servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java [hunk 1]
```diff
@@ -406,6 +396,67 @@ public <R> R getCommitLog(
...
+        Operation.Delete delete = (Operation.Delete) op;
+        check = canReadContentKey(endRef.getValue(), delete.getKey(), null);
+      } else {
+        throw new IllegalStateException("Unknown operation " + op);
+      }
+      if (failures.containsKey(check)) {
+        failedChecks.add(check);
+      } else if (!failedChecks.contains(check)) {
+        if (successfulChecks.contains(check)) {
+          newLogEntry.addOperations(op);
```
Comment (dimas-b) [inline servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java:450]: This can probably be outside the `if` (also line 452)

**Review by dimas-b — COMMENTED** [review]
LGTM with some rather minor comments.

File: versioned/spi/src/main/java/org/projectnessie/versioned/paging/PaginationIterator.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
 
   String tokenForCurrent();
 
+  String tokenForEntry(T entry);
```
[the remark this answers is not in the document]
Comment (snazy) [reply versioned/spi/src/main/java/org/projectnessie/versioned/paging/PaginationIterator.java:25]: yes

[the remark this answers is not in the document]
Comment (dimas-b) [reply versioned/spi/src/main/java/org/projectnessie/versioned/paging/PaginationIterator.java:25]: what's the difference between `tokenForCurrent()` and `tokenForEntry(next())`?

**Review by dimas-b — APPROVED** [review]

File: versioned/spi/src/main/java/org/projectnessie/versioned/paging/PaginationIterator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (snazy) [reply versioned/spi/src/main/java/org/projectnessie/versioned/paging/PaginationIterator.java:25]: current != next ;)

**Pull request merged**